fre/make/gfdlfremake/buildDocker.py

```diff
@@ -187,14 +187,35 @@ def writeRunscript(self,RUNenv,containerRun,runOnDisk):
     def build(self,containerBuild,containerRun):
         """
         Brief: Builds the container image for the model
-        Param: 
+        Param:
             - self : The dockerfile object
-            - containerBuild : The tool used to build the container; 
+            - containerBuild : The tool used to build the container;
                                docker or podman used
-            - containerRun : The container platform used with `exec` to 
+            - containerRun : The container platform used with `exec` to
                              run the container; apptainer or singularity used
         """
         os.system(containerBuild+" build -f Dockerfile -t "+self.e+":"+self.target.gettargetName())
         os.system("rm -f "+self.e+".tar "+self.e+".sif")
         os.system(containerBuild+" save -o "+self.e+"-"+self.target.gettargetName()+".tar localhost/"+self.e+":"+self.target.gettargetName())
         os.system(containerRun+" build --disable-cache "+self.e+"-"+self.target.gettargetName()+".sif docker-archive://"+self.e+"-"+self.target.gettargetName()+".tar")
+
+    def createBuildScript(self,containerBuild,containerRun):
```

Comment:
Great idea, thanks for this! Takes a little bit more of the magic of how to build the actual container away. 🎉 Maybe we should actually make the creation of this script a default behavior in `def build`. (basically replacing `def build` with `def createBuildScript`) 
Then in `runFremake.py`, instead of 
```
if execute:
    dockerBuild.build(containerBuild,containerRun)
else:
    dockerBuild.createBuildScript(containerBuild, containerRun)
    print("Container build script created at "+dockerBuild.userScriptPath+"\n\n")
```
It would be:
```
#create the container build script
dockerBuild.createBuildScript(containerBuild, containerRun)

#print statement 
(ADD print statement somewhere about where the script is located)

if execute:
    subprocess.run(args=[dockerBuild.userScriptPath], check=True)
else:
    print("Container build script created at "+dockerBuild.userScriptPath+"\n\n")
```

What do you think? Or do you think it might be better to have one less script being made if it doesn't need to be (i.e. if execute is passed, no need to make the build script)

Reply:
Yeah I think always making the script is a good idea just cause it'll be a little more transparent and easier to debug build issues. I'll add those changes in!